wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -135,11 +135,9 @@ abstract class Wallet
   override def processCompactFilters(
       blockFilters: Vector[(DoubleSha256Digest, GolombFilter)]): Future[
     Wallet] = {
-    val utxosF = listUtxos()
-    val addressesF = listAddresses()
     for {
-      utxos <- utxosF
-      addresses <- addressesF
+      utxos <- listUtxos()
```

Comment:
This is a performance regression for #1282 / #1283

Reply:
Also according to sqlite you can safely read concurrently from a sqlite database, however writing is where the problem comes into play if I understand @rorp correctly

https://stackoverflow.com/a/4060838/967713

Reply:
That fixes potential SQLITE_BUSY on production size databases. To fix the bottleneck (if it actually exists) the join should be done on the database level.

Reply:
Since we now process filters in batches, it shouldn't be as much a concern

Reply:
>That fixes potential SQLITE_BUSY on production size databases. To fix the bottleneck (if it actually exists) the join should be done on the database level.

SQLITE_BUSY only occurs on writes, there is no point in reducing performance for reads: https://sqlite.org/rescode.html#busy